addons/mrp_subcontracting/models/stock_picking.py

```diff
@@ -147,17 +147,33 @@ def _prepare_subcontract_mo_vals(self, subcontract_move, bom):
 
     def _subcontracted_produce(self, subcontract_details):
         self.ensure_one()
+
+        group_move = {}
+        group_by_company = defaultdict(list)
         for move, bom in subcontract_details:
             if float_compare(move.product_qty, 0, precision_rounding=move.product_uom.rounding) <= 0:
                 # If a subcontracted amount is decreased, don't create a MO that would be for a negative value.
                 continue
-            mo = self.env['mrp.production'].with_company(move.company_id).create(self._prepare_subcontract_mo_vals(move, bom))
-            self.env['stock.move'].create(mo._get_moves_raw_values())
-            self.env['stock.move'].create(mo._get_moves_finished_values())
-            mo.date_planned_finished = move.date  # Avoid to have the picking late depending of the MO
```

Comment:
This one disappear. Are you sure it won't break anything

Reply:
From @Whenrow:
> @auma-odoo I just checked, and it seems we don't need to rewrite the date after the creation anymore. The line was introduced in [326ae2a](https://github.com/odoo/odoo/commit/326ae2a725715fa7e03f6675b9de26a5438fbf10), but commit [94cb5b6](https://github.com/odoo/odoo/commit/94cb5b6e8f9f932b0bbbbe76afa60e9dd93456fe) makes it useless. You can remove it :)